libkbfs/block_getter.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+
+	"golang.org/x/net/context"
+)
+
+// blockGetter provides the API for the block retrieval worker to obtain blocks.
+type blockGetter interface {
+	getBlock(context.Context, KeyMetadata, BlockPointer, Block) error
+}
+
+// realBlockGetter obtains real blocks using the APIs available in Config.
+type realBlockGetter struct {
+	config Config
+}
+
+// getBlock implements the interface for realBlockGetter.
+func (bg *realBlockGetter) getBlock(ctx context.Context, kmd KeyMetadata, blockPtr BlockPointer, block Block) error {
```

Comment:
Per my comment below, might also have to change this function to take a `blockType` enum and return a `Block`, as well as `DecryptBlock`.